tests/v2/validation/provisioning/rke2/agent_customization_test.go

```diff
@@ -0,0 +1,244 @@
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(r.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(r.T(), err)
+
+	r.standardUserClient = standardUserClient
+}
+
+func (r *RKE2AgentCustomizationTestSuite) TestProvisioningRKE2ClusterAgentCustomization() {
+	productionPool := []provisioninginput.MachinePools{
+		{
+			NodeRoles: machinepools.NodeRoles{
+				ControlPlane: true,
+				Etcd:         false,
+				Worker:       false,
+				Quantity:     2,
+			},
+		},
+		{
+			NodeRoles: machinepools.NodeRoles{
+				ControlPlane: false,
+				Etcd:         true,
+				Worker:       false,
+				Quantity:     3,
+			},
+		},
+		{
+			NodeRoles: machinepools.NodeRoles{
+				ControlPlane: false,
+				Etcd:         false,
+				Worker:       true,
+				Quantity:     2,
+			},
+		},
+	}
+
+	agentCustomization := management.AgentDeploymentCustomization{
+		AppendTolerations: []management.Toleration{
+			{
+				Key:   "TestKeyToleration",
+				Value: "TestValueToleration",
+			},
+		},
+		OverrideResourceRequirements: &management.ResourceRequirements{
+			Limits: map[string]string{
+				"cpu": "750m",
+				"mem": "500Mi",
+			},
+			Requests: map[string]string{
+				"cpu": "250m",
+			},
+		},
+		OverrideAffinity: &management.Affinity{
+			NodeAffinity: &management.NodeAffinity{
+				PreferredDuringSchedulingIgnoredDuringExecution: []management.PreferredSchedulingTerm{
+					{
+						Preference: &management.NodeSelectorTerm{
+							MatchExpressions: []management.NodeSelectorRequirement{
+								{
+									Key:      "testAffinityKey",
+									Operator: "In",
+									Values:   []string{"true"},
+								},
+							},
+						},
+						Weight: 100,
+					},
+				},
+			},
+		},
+	}
+
+	customAgents := []string{"fleet-agent", "cluster-agent"}
+	tests := []struct {
+		name         string
+		machinePools []provisioninginput.MachinePools
+		client       *rancher.Client
+		agent        string
+	}{
+		{"Custom Fleet Agent - Standard User", productionPool, r.standardUserClient, customAgents[0]},
+		{"Custom Cluster Agent - Standard User", productionPool, r.standardUserClient, customAgents[1]},
+	}
+
+	for _, tt := range tests {
+		subSession := r.session.NewSession()
+		defer subSession.Cleanup()
+
+		client, err := tt.client.WithSession(subSession)
+		require.NoError(r.T(), err)
+		r.provisioningConfig.MachinePools = tt.machinePools
+
+		if tt.agent == "fleet-agent" {
+			r.provisioningConfig.FleetAgent = &agentCustomization
+			r.provisioningConfig.ClusterAgent = nil
+		}
+
+		if tt.agent == "cluster-agent" {
+			r.provisioningConfig.ClusterAgent = &agentCustomization
+			r.provisioningConfig.FleetAgent = nil
+		}
+
+		permutations.RunTestPermutations(&r.Suite, tt.name, client, r.provisioningConfig, permutations.RKE2ProvisionCluster, nil, nil)
+	}
+}
+
+func (r *RKE2AgentCustomizationTestSuite) TestProvisioningRKE2NegativeClusterAgentCustomization() {
```

Comment:
Maybe we can change this name? I get the `negative` part, but maybe something like `TestFailedProvisioningRKE2ClusterAgentCustomization`?

Reply:
went with `failure` instead of `failed` if that's okay :)